### ISSUE TITLE
Problem connecting buffer

### ISSUE THREAD

**Issue by falematte** [millejoh/emacs-ipython-notebook#307]
Whenever I connect a buffer (usually originating from a babel block by the C-c ' command) to a notebook and I press C-c C-c nothing happen except the following message:

`ein: [info] Command sent to the kernel: %run /home/.../file.py`

Any tips?

**Comment by millejoh** [comment]
You should not be calling `ein:connect-to-notebook-buffer` inside an org source edit buffer. This will overwrite some of the keybindings and lead to the behavior you note above. EIN takes some steps to associate a source edit buffer with a running kernel, mostly so autocompletion will work. If there is something in particular behavior you are looking for let me know; I will see if I can add it.

**Comment by falematte** [comment]
Thank you for the answer. Essentially I am using your software instead of ob-ipython. Therefore I am using only babel blocks (ein blocks). It would be very nice to be able to execute a block from the narrowed buffer with C-c C-c...

Off-topic: Another very interesting feature would be to be able to copy a babel block automatically into a jupyter notebook and vice-versa.

**Comment by millejoh** [comment]
My bad - I'd forgotten how this code works. Turns out ein calls `ein:connect-to-notebook-buffer` as part of the creation of an org source edit buffer, which completely explains the behavior you are seeing.

I've added a small hack in the latest commit to implement the behavior you are looking for. See the function `org-babel-edit:execute-ein`, which is now bound to `C-c C-c`.

Let me know how it works.
